From 04471cc864fb87af1785d32016c33d816337eb52 Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Thu, 2 Aug 2018 15:41:00 +0800
Subject: [PATCH 4288/5242] MLK-19116 ARMR64: dts: imx8qm mek root: disable
 smmu and change sid for usdhc1

commit  f04477c6c8c805b8b587ad104df5e20631ffca29 from
https://source.codeaurora.org/external/imx/linux-imx.git

SMMU is being managed by Jailhosue hypervisor, so disable it in
root cell.

SDHC1 is owned by non-root cell Linux, but we need U-Boot
to program SID according to the iommus property.

Since SDHC1 is in different cell, it need a different SID to avoid DMA
attack.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
(cherry picked from commit 5ac1850f8054874ef69d2c26cd777c37533627d1)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-mek-root.dts     |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-root.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-root.dts
index 98ca22a..e28fd5f 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-root.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-root.dts
@@ -102,9 +102,15 @@
 	};
 };
 
+&smmu {
+	/* Jailhouse hypervisor will initialize SMMU and use it. */
+	status = "disabled";
+};
+
 &usdhc1 {
+	/* Let U-Boot program SID */
+	iommus = <&smmu 0x10 0x7f80>;
 	/delete-property/ compatible;
-	/delete-property/ iommus;
 };
 
 &lpuart2 {
-- 
1.7.9.5

